src/V3Timing.cpp

```diff
@@ -0,0 +1,785 @@
+            senItemsp
+                = VN_AS(AstNode::addNext(senItemsp, new AstSenItem{flp, VEdgeType::ET_CHANGED,
+                                                                   refp->cloneTree(false)}),
+                        SenItem);
+        });
+        // Create an always with the assignment under it (assign the RHS to the intermediate)
+        auto* const alwaysp
+            = new AstAlways{flp, VAlwaysKwd::ALWAYS, new AstSenTree{flp, senItemsp},
+                            new AstAssign{flp, new AstVarRef{flp, newvscp, VAccess::WRITE},
+                                          nodep->rhsp()->unlinkFrBack()}};
+        alwaysp->user1SetOnce();  // Do not revisit this always, this one can be run many times
+                                  // concurrently
+        controlp->replaceWith(alwaysp);
+        alwaysp->addStmtp(controlp);
+        // Assign the intermediate to the LHS after the timing control
+        auto* const assignp = new AstAssign{flp, nodep->lhsp()->unlinkFrBack(),
+                                            new AstVarRef{flp, newvscp, VAccess::READ}};
+        assignp->user1SetOnce();  // So we don't do the net delay again
+        nodep->replaceWith(assignp);
+        nodep->deleteTree();
+    }
+    virtual void visit(AstAssignDly* nodep) override {
+        iterateChildren(nodep);
+        auto* const flp = nodep->fileline();
+        auto* const controlp = factorOutTimingControl(nodep);
+        if (!m_procp->user2() && !controlp)
+            return;  // Leave if not suspendable and no timing control
+        // Handle the intra assignment timing control (or transform if we're in a suspendable)
+        // Create an intermediate var
+        auto* const newvscp
+            = m_scopep->createTemp(m_dlyVarNames.get(nodep->lhsp()), nodep->lhsp()->width());
+        // Add a bool var that says if we want to commit this NBA
+        auto* const chgVscp = m_scopep->createTemp(m_dlyChgVarNames.get(nodep), 1);
+        m_procp->addNextHere(new AstInitialStatic{
+            flp, new AstAssign{flp, new AstVarRef{flp, chgVscp, VAccess::WRITE},
+                               new AstConst{flp, AstConst::BitFalse()}}});
+        // Create an AlwaysPost that will be responsible for committing it
+        auto* const ifp = new AstIf{flp, new AstVarRef{flp, chgVscp, VAccess::READ},
+                                    new AstAssignDly{flp, nodep->lhsp()->unlinkFrBack(),
+                                                     new AstVarRef{flp, newvscp, VAccess::READ}}};
+        ifp->addIfsp(new AstAssign{flp, new AstVarRef{flp, chgVscp, VAccess::WRITE},
+                                   new AstConst{flp, AstConst::BitFalse()}});
+        auto* const alwaysp = new AstAlwaysPost{
+            flp,
+            new AstSenTree{flp, new AstSenItem{flp, VEdgeType::ET_CHANGED,
+                                               new AstVarRef{flp, chgVscp, VAccess::READ}}},
+            ifp};
```

Comment:
This will create a new trigger for each NBA, which is too many. The SenTree of this AlwaysPost should be the union of all SenTrees that can set the commit flag

Reply:
Fixed, but by making it combo. Otherwise getting all the necessary sentrees would require duplicating the timing sentree gathering logic from `V3Timing`